vpr/src/route/bucket.h

```diff
@@ -167,6 +167,13 @@ class Bucket : public HeapInterface {
         t_heap* hptr = &items_.alloc_item()->item;
         return hptr;
     }
+
+    t_heap* alloc(bool init_data_structs) {
+        // Supress unused warning
+        (void)init_data_structs;
+        return alloc();
+    }
```

Comment:
Correct, `alloc(bool init_data_structs)` is a function I only implemented for the binary_heap class so far. The intention is to allow the router to selectively create  t_heap_path structure in t_heap when RCV is turned on, and skip that and create a nullptr when it's turned off for performance reasons. 

I haven't implemented this for bucket heaps yet which would cause issues when run with RCV! I'll rename the `init_data_structs` variable, comment, and do you think it would be a good idea to cause a hard exception if this function is used, for the time being?

Reply:
Yes, it buckets heaps don't support RCV (which is OK) that should be commented and you should give a fatal error (assert or VPR_ERROR) if someone tries to use RCV with bucket heaps.

Reply:
I think the design choice to allocate the additional heap data with a boolean and as a raw new / delete should be discussed.  As far as I can tell, it doesn't really belong inside of HeapStorage at all.  Given what `path_data` is doing, I believe I would advocate that `path_data` be a `StrongId<PathDataTag>`, and create a data management class for dealing with the path data.  Currently the connection router is littered with `path_data` management, but it appears to be something that is maybe better centerlized outside of the connection router to being with?